tests/unit/test_run.py

```diff
@@ -1229,8 +1229,70 @@ def test_median_over(test_processing_scm_df):
     pd.testing.assert_frame_equal(exp.set_index(obs.index.names), obs, check_like=True)
 
 
+def test_arb_function_over(test_processing_scm_df):
+    def same(df):
+        return df
+
+    obs = test_processing_scm_df.process_over("scenario", same)
+
+    pd.testing.assert_frame_equal(
+        test_processing_scm_df.timeseries(), obs, check_like=True
+    )
+
+    def add_2(df):
+        return df + 2
+
+    obs = test_processing_scm_df.process_over("scenario", add_2)
+    pd.testing.assert_frame_equal(
+        (test_processing_scm_df + 2).timeseries(), obs, check_like=True
+    )
+
+
+def test_arb_function_returns_none(test_processing_scm_df):
+    def add_2_only_coal(df):
+        variable = df.index.get_level_values("variable").unique()[0]
+        if variable == "Primary Energy|Coal":
+            return df + 2
+        # implicit return None for any other variables
```

Comment:
so this means the group gets dropped? If yes, cool.

Reply:
Yes, the out doesn't contain any results for groups which return None